src/lighteval/tasks/templates/continuation.py

```diff
@@ -0,0 +1,159 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Callable
+
+from typing_extensions import NotRequired, TypedDict
+
+from lighteval.tasks.requests import Doc
+from lighteval.tasks.templates.utils.formatting_utils import (
+    capitalize,
+    fix_capitalization,
+    fix_ending_punct,
+    is_ended_sentence,
+)
+from lighteval.tasks.templates.utils.formulation import (
+    CFFormulation,
+    Formulation,
+    MCFFormulation,
+    build_answers,
+    build_options,
+)
+from lighteval.tasks.templates.utils.translation_literals import TRANSLATION_LITERALS
+from lighteval.tasks.templates.utils.utils import create_adapter_from_dict
+from lighteval.utils.language import Language
+from lighteval.utils.utils import as_list
+
+
+CONTINUATION_QUERY_CF = "{instruction}{context}"
+
+CONTINUATION_QUERY_MCF = "{instruction}{context}\n{options}{answer_word}{colon}"
+
+
+# NO idea how to ensure we have the same keys in these typedicts in python :(
+class ContinuationInput(TypedDict):
+    context: str
+    continuations: list[str]
+    gold_idx: list[int] | int
+    instruction: NotRequired[str]
+
+
+class ContinuationDictAdapter(TypedDict):
+    context: str
+    continuations: str
+    gold_idx: str
+    instruction: NotRequired[str]
+
+
+# Python too dumb to do fancy inference :(
+
+
+def get_continuation_prompt_function(
+    language: Language,
+    adapter: Callable[[dict], ContinuationInput] | ContinuationDictAdapter,
+    formulation: Formulation = MCFFormulation(),
+):
+    """
+    Create a templated prompt function for a Continuation task.
+    Example tasks:
+    - Hellaswag
+    - XStoryCloze
+
+    Format:
+    Context xxx | Continuation 1 | Continuation 2 | Continuation 3
+
+    Args:
+        language (Language): The language of the Continuation task.
+        adapter (Callable[[dict], ContinuationInput] | ContinuationDictAdapter): A function or dictionary to adapt the input data to the required ContinuationInput format.
+            Must map data from the dataset row to the ContinuationInput format.
+            Note: The gold_idx must be an index or list of indices in the continuations list, indicating the correct continuation(s).
+        formulation (Formulation, optional): The formulation to use for the task. Defaults to MCFFormulation().
+    Returns:
+        Callable: A function that generates Continuation prompts based on the given parameters.
+    """
+    adapter_fn: Callable[[dict], ContinuationInput] = (
+        create_adapter_from_dict(adapter) if isinstance(adapter, dict) else adapter  # type: ignore
```

Comment:
If you edit `create_adapter_from_dict` to an `as_adapter` function (similar to the `as_list` we have), it could take a dict, adapter or other, and return an adapter or raise an error message - it would simplify your code all the times you call the function